demos/python_demos/common/images_capture.py

```diff
@@ -0,0 +1,176 @@
+        super().__init__(loop)
+        if not os.path.isfile(input):
+            raise InvalidInput("Can't find the image by {}".format(input))
+        self.image = cv2.imread(input, cv2.IMREAD_COLOR)
+        if self.image is None:
+            raise OpenError("Can't open the image from {}".format(input))
+        self.can_read = True
+
+    def read(self):
+        if self.loop:
+            return copy.deepcopy(self.image)
+        if self.can_read:
+            self.can_read = False
+            return copy.deepcopy(self.image)
+        return None
+
+    def fps(self):
+        return 1.0
+
+    def get_type(self):
+        return 'IMAGE'
+
+
+class DirReader(ImagesCapture):
+
+    def __init__(self, input, loop):
+        super().__init__(loop)
+        self.dir = input
+        if not os.path.isdir(self.dir):
+            raise InvalidInput("Can't find the dir by {}".format(input))
+        self.names = sorted(os.listdir(self.dir))
+        if not self.names:
+            raise OpenError("The dir {} is empty".format(input))
+        self.file_id = 0
+        for name in self.names:
+            filename = os.path.join(self.dir, name)
+            image = cv2.imread(filename, cv2.IMREAD_COLOR)
+            if image is not None:
+                return
+        raise OpenError("Can't read the first image from {}".format(input))
+
+    def read(self):
+        while self.file_id < len(self.names):
+            filename = os.path.join(self.dir, self.names[self.file_id])
+            image = cv2.imread(filename, cv2.IMREAD_COLOR)
+            self.file_id += 1
+            if image is not None:
+                return image
+        if self.loop:
+            self.file_id = 0
+            while self.file_id < len(self.names):
+                filename = os.path.join(self.dir, self.names[self.file_id])
+                image = cv2.imread(filename, cv2.IMREAD_COLOR)
+                self.file_id += 1
+                if image is not None:
+                    return image
+        return None
+
+    def fps(self):
+        return 1.0
+
+    def get_type(self):
+        return 'DIR'
+
+
+class VideoCapWrapper(ImagesCapture):
+
+    def __init__(self, input, loop):
+        super().__init__(loop)
+        self.cap = cv2.VideoCapture()
+        status = self.cap.open(input)
+        if not status:
+            raise InvalidInput("Can't open the video from {}".format(input))
+
+    def read(self):
+        status, image = self.cap.read()
+        if not status:
+            if not self.loop:
+                return None
+            self.cap.set(cv2.CAP_PROP_POS_FRAMES, 0)
+            status, image = self.cap.read()
+            if not status:
+                return None
+        return image
+
+    def fps(self):
+        return self.cap.get(cv2.CAP_PROP_FPS)
+
+    def get_type(self):
+        return 'VIDEO'
+
+
+class CameraCapWrapper(ImagesCapture):
+
+    def __init__(self, input, loop, camera_resolution):
```

Comment:
> loop won't work there, so it could be erased too.

Init shouldn't take loop now. You need to either initialize `ImagesCapture`'s loop with `False` or remove this field from the base class.

Reply:
Remove it.